core/discovery/server.go

```diff
@@ -81,6 +87,13 @@ func (d *discoveryServer) checkMetadataEditRights(ctx context.Context, in *pb.Me
 			return errPermissionDeniedf(`No "%s" rights to Application "%s"`, rights.AppDelete, appID)
 		}
 	}
+
+	// Check claims for GatewayID
+	if gatewayID != "" {
+		if !claims.AppRight(gatewayID, rights.GatewayDelete) {
```

Comment:
- Use `claims.GatewayRight` instead of `claims.AppRight` (otherwise the wrong token scope is checked)
- Add `|| (claims.Type == "gateway" && claims.Subject == gatewayID)` (the gateway might announce itself too)